File: calculator/src/main/java/com/turlir/calculator/member/OperatorPrint.java
package com.turlir.calculator.member;


import com.turlir.calculator.converter.Printer;
import com.turlir.calculator.converter.Visual;

class OperatorPrint implements Visual {

    private final String mToken;

    OperatorPrint(String token) {
        mToken = token;
    }

    @Override
    public void print(Printer chain) {
        chain.append(mToken);
    }

    @Override
    public String toString() {
        return mToken;
    }
}

File: calculator/src/main/java/com/turlir/calculator/member/OperatorPrint.kt
package com.turlir.calculator.member


import com.turlir.calculator.converter.Printer
import com.turlir.calculator.converter.Visual

internal class OperatorPrint(private val mToken: String) : Visual {

    override fun print(chain: Printer) {
        chain.append(mToken)
    }

    override fun toString(): String {
        return mToken
    }
}
